### ISSUE TITLE
2019.2.5f1: No UI buttons, no gizmos, unusable

### ISSUE THREAD

**Issue by JillCrungus** [sabresaurus/SabreCSG#233]
SabreCSG is completely unusable in Unity 2019.2.5f1, the UI does not show and nor do any gizmos show on CSGModels.

![image](https://user-images.githubusercontent.com/32760843/67231329-0671a300-f437-11e9-9f06-5f0aebb60da1.png)

The console warns that SceneView.onSceneGUIDelegate is obsolete.
![image](https://user-images.githubusercontent.com/32760843/67231376-1c7f6380-f437-11e9-96ce-9c8ddafa3f02.png)

**Comment by pjbaron** [comment]
Same in 2019.2.10f1

I'm looking at the errors and they're mostly trivial to fix but some parts like this:
```if (!EditorHelper.SceneViewHasDelegate(OnSceneGUI))``` (CSGModel.cs:1345) inspects the list of delegates for the SceneView.  That doesn't appear to be possible (the new 'SceneView.duringSceneGui' can only be used on the left of a += or -= instruction.

This was always a bit hacky, the comment explains:
```            // Make sure the events we need to listen for are all bound (recompilation removes listeners, so it is
            // necessary to rebind dynamically)```

So maybe someone can answer the question: in Unity 2019 does recompilation still remove listeners?
If not then we can just add the delegate once in a constructor or init function.

If that won't work then (unless SceneView contains a way to check if a delegate has already been assigned - I flicked through the docs but didn't see one) we will need to maintain a list of which delegates were added, and reset it when compilation occurs.
